nomos-cli/src/da/disseminate.rs

```diff
@@ -1,76 +1,79 @@
-use crate::api::mempool::send_certificate;
-use clap::{Args, ValueEnum};
-use full_replication::{AbsoluteNumber, Attestation, Certificate, FullReplication, Voter};
-use futures::StreamExt;
-use hex::FromHex;
-use nomos_core::wire;
-use nomos_da::network::{adapters::libp2p::Libp2pAdapter as DaNetworkAdapter, NetworkAdapter};
+use clap::Args;
+use kzgrs_backend::{
+    common::build_attestation_message,
+    dispersal::{BlobInfo, Metadata},
+    encoder::EncodedData as KzgEncodedData,
+};
+use nomos_core::{
+    da::{DaDispersal, DaEncoder},
+    wire,
+};
+use nomos_da_network_service::{
+    backends::mock::executor::MockExecutorBackend as NetworkBackend, NetworkService,
+};
 use nomos_log::Logger;
-use nomos_network::backends::libp2p::Libp2p as NetworkBackend;
-use nomos_network::NetworkService;
 use overwatch_derive::*;
 use overwatch_rs::{
     services::{
         handle::{ServiceHandle, ServiceStateHandle},
-        relay::NoMessage,
+        relay::{NoMessage, OutboundRelay, Relay},
         state::*,
         ServiceCore, ServiceData, ServiceId,
     },
     DynError,
 };
 use reqwest::Url;
-use serde::Serialize;
 use std::{
-    error::Error,
     path::PathBuf,
     sync::{mpsc::Sender, Arc},
     time::Duration,
 };
 use tokio::sync::{mpsc::UnboundedReceiver, Mutex};
 
-pub async fn disseminate_and_wait<D, B, N, A, C>(
-    mut da: D,
+use super::network::adapters::mock::MockExecutorDispersalAdapter;
+use crate::api::mempool::send_blob_info;
+
+const APP_ID: &[u8; 32] = b"1234567890abcdef1234567890abcdef";
+
+pub async fn disseminate_and_wait<E, D>(
+    encoder: &E,
+    disperal: &D,
     data: Box<[u8]>,
-    adapter: N,
     status_updates: Sender<Status>,
     node_addr: Option<&Url>,
     output: Option<&PathBuf>,
 ) -> Result<(), Box<dyn std::error::Error + Send + Sync>>
 where
-    D: DaProtocol<Blob = B, Attestation = A, Certificate = C>,
-    N: NetworkAdapter<Blob = B, Attestation = A> + Send + Sync,
-    C: Serialize,
+    E: DaEncoder<EncodedData = KzgEncodedData>,
+    D: DaDispersal<EncodedData = KzgEncodedData>,
+    <E as nomos_core::da::DaEncoder>::Error: std::error::Error + Send + Sync + 'static,
+    <D as nomos_core::da::DaDispersal>::Error: std::error::Error + Send + Sync + 'static,
 {
     // 1) Building blob
     status_updates.send(Status::Encoding)?;
-    let blobs = da.encode(data);
+    let encoded_data = encoder.encode(&data).map_err(Box::new)?;
+    let blob_hash = build_attestation_message(
+        &encoded_data.aggregated_column_commitment,
+        &encoded_data.row_commitments,
+    );
```

Comment:
We do not have attestation per se now. We should probably rename this and adapt to latest version of the protocol. Do you mind opening an issue for this?

Reply:
I'll rename it here to `build_blob_id` (if this the name sounds good).

Reply:
`build_blob_info`?`

Reply:
`blob_info` is `{blob_id + metadata}`. As the result of this function is hash, maybe `build_blob_hash`?

Reply:
`build_blob_id` works then. Sorry, I mixed the call with other stuff.